src/lib/Types/Core.hs

```diff
@@ -264,16 +268,19 @@ data ClassDef (n::S) where
   ClassDef
     :: SourceName
     -> [SourceName]              -- method source names
-    -> Nest Binder n1 n2         -- parameters
+    -> Nest RoleBinder n1 n2     -- parameters
```

Comment:
Why do typeclasses have role binders too? I thought it's just a datacon thing?

Reply:
I expected it to be just a type constructor thing too, but it turned out that I needed them for classes and instances too. It makes sense for classes because they're a lot like a type constructor (and indeed `Ix n` is a type). Concretely we need them to handle the `DictTy` case of `traverseTyParams`. Instances need the roles for a difference reason, which is that the roles tell you how to generalize a particular dict derivation to a more general type.